core/src/main/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitor.scala

```diff
@@ -476,7 +476,7 @@ private[spark] class ExecutorMonitor(
     }
 
     def removeShuffle(id: Int): Unit = {
-      if (shuffleIds.remove(id) && shuffleIds.isEmpty) {
+      if (shuffleIds != null && shuffleIds.remove(id) && shuffleIds.isEmpty) {
```

Comment:
I think it would be better to modify `shuffleCleaned` to not post anything if the tracking feature is disabled.

Reply:
(And verify in the test that nothing was posted.)

Reply:
Yea, that would work too. Let me fix it like that.